docs/v3/group-chat.md

```diff
@@ -0,0 +1,165 @@
+
+In some cases, LibXMTP will perform a sync automatically. This currently includes calls to add/remove members to a group or using the stream functions. Outside of those methods, apps can call `sync()` when they choose in order to efficiently manage network traffic.
+
+## Group Protocol
+
+The group protocol contains methods specific to individual groups, identified by their `group_id`. These methods allow a user to communicate with the group. Each SDK has their own version of this protocol, using the underlying implementation in [LibXMTP](https://github.com/xmtp/libxmtp).
+
+| Method | Description |
+| ---- | ---- |
+| send | Send a message to this group. Requires sync. |
+| sync | Query for new messages, proposals, and commits, and send unpublished messages related to this group. |
+| find_messages | Query the database for stored messages, with filters for time frame and limit. Requires sync.|
+| list_members | A list including the account address and installation id of each member. Requires sync. |
+| add_members | Add multiple group members by account address. Syncs automatically.|
+| remove_members | Remove multiple group members by account address. Syncs automatically.|
+| stream | Stream messages to a callback function. Syncs automatically.|
+| created_at_ns | Group creation time. |
+| is_active | Whether the group is active. Requires sync. |
+| added_by_address | The address of the group member who added this user to the group. |
+| group_metadata | Group metadata, such as group title and permissions. Requires sync.|
+
+### Delivery Status
+
+When a message is sent, it is stored in the local database with a status of `Unpublished`. `find_messages()` will display unpublished messages by default unless you filter on `DeliveryStatus`. Unpublished messages can be displayed to the user but they should be marked as not yet sent in the app UI.
+
+When the app calls `sync()`, LibXMTP will make three attempts to automatically send the message to the network. If all three attempts fail, the message status will be marked as `Failed`. If it succeeds, it will be marked as `Published`.
+
+## Spam Protection
+
+To prevent unwanted contacts, the app should apply the existing [Universal allow/block Preferences](../build/user-consent.md) to group chats. The Group protocol provides the `added_by_address` method to find out who has initiated the user into a group chat. Apps can compare this address using the standard `isAllowed()` or `isDenied()` functions on the contacts to determine how and whether the group should be listed, and how and whether messages are displayed, based on the design of the app and the user's settings.
```

Comment:
No longer find by address but instead `added_by_inbox_id` and we have new consent that checks if the inbox Id is allowed or denied. See https://github.com/xmtp/xmtp-android/pull/244